pkg/driver/opcua/device.go

```diff
@@ -136,6 +136,11 @@ func (d *device) handleTraitEvent(ctx context.Context, node *ua.NodeID, value an
 	}
 }
 
+// deviceNameField returns a zap field for the device name, ensuring consistent logging across all device operations.
+func (d *device) deviceNameField() zap.Field {
+	return zap.String("device", d.conf.Name)
+}
+
```

Comment:
Not what I meant. d.logger can be assigned `d.logger = logger.With(zap.String("device"...))` on construction